dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -81,24 +94,64 @@ object Datadog {
                 InternalLogger.Target.USER,
                 CANNOT_CREATE_SDK_INSTANCE_ID_ERROR
             )
-            return
+            return null
         }
 
-        globalSdkCore = DatadogCore(context, credentials, configuration, sdkInstanceId)
-        globalSdkCore.setTrackingConsent(trackingConsent)
+        val sdkCore = DatadogCore(context, credentials, configuration, sdkInstanceId)
+        sdkCore.setTrackingConsent(trackingConsent)
+        registry.register(instanceName, sdkCore)
 
-        initialized.set(true)
+        // TODO RUMM-3103 remove this
+        if (instanceName == null) {
+            globalSdkCore = sdkCore
+        }
+        return sdkCore
+    }
+
+    /**
+     * Initializes the Datadog SDK.
+     * @param context your application context
+     * @param credentials your organization credentials
+     * @param configuration the configuration for the SDK library
+     * @param trackingConsent as the initial state of the tracking consent flag
+     * @return the initialized SDK instance, or null if something prevents the SDK from
+     * being initialized
+     * @see [Credentials]
+     * @see [Configuration]
+     * @see [TrackingConsent]
+     * @throws IllegalArgumentException if the env name is using illegal characters and your
+     * application is in debug mode otherwise returns false and stops initializing the SDK
+     */
+    @JvmStatic
+    fun initialize(
+        context: Context,
+        credentials: Credentials,
+        configuration: Configuration,
+        trackingConsent: TrackingConsent
+    ): SdkCore? {
+        return initialize(null, context, credentials, configuration, trackingConsent)
```

Comment:
I think we should give a name to the default instance. In the end it will be useful for internal logging, for example, where logs can come from multiple instances and it is need to distinguish them.

Upd: Ah, I see the default one is in the `SdkCoreRegistry`.